enterprise/reporting/client/client.go

```diff
@@ -93,9 +94,43 @@ func New(path Route, conf *config.Config, log logger.Logger, stats stats.Stats)
 	reportingServiceURL := conf.GetString("REPORTING_URL", "https://reporting.dev.rudderlabs.com")
 	reportingServiceURL = strings.TrimSuffix(reportingServiceURL, "/")
 
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+
+	if conf.IsSet("Reporting.httpClient.transport.maxIdleConns") {
+		transport.MaxIdleConns = conf.GetInt("Reporting.httpClient.transport.maxIdleConns", 100)
```

Comment:
note: On this use case given on a single host use, maxIdleConns should not be effective.

 https://github.com/golang/go/blob/master/src/net/http/transport.go#L211